fog/types/src/view.rs

```diff
@@ -342,18 +374,34 @@ pub struct FogTxOut {
 
 // Convert a TxOut to a FogTxOut in the efficient way (omitting compressed
 // commitment)
-impl From<&TxOut> for FogTxOut {
+impl TryFrom<&TxOut> for FogTxOut {
+    type Error = FogTxOutError;
+
     #[inline]
-    fn from(src: &TxOut) -> Self {
-        Self {
-            target_key: src.target_key,
-            public_key: src.public_key,
-            amount_masked_value: src.masked_amount.masked_value,
-            amount_masked_token_id: src.masked_amount.masked_token_id.clone(),
-            amount_commitment_data_crc32: Crc::<u32>::new(&crc::CRC_32_ISO_HDLC)
-                .checksum(src.masked_amount.commitment.point.as_bytes()),
-            e_memo: src.e_memo,
-        }
+    fn try_from(src: &TxOut) -> Result<Self, Self::Error> {
+        let (amount_masked_value, amount_masked_token_id, amount_commitment_data_crc32) = match src.get_masked_amount().map_err(|_| FogTxOutError::Amount(AmountError::MissingMaskedAmount))? {
+                VersionedMaskedAmount::V1(masked_amount) => {
+                    (masked_amount.masked_value,
+                    MaskedTokenId::V1(
+                        masked_amount.masked_token_id.clone(),
+                    ), masked_amount.commitment_crc32())
+                },
+                VersionedMaskedAmount::V2(masked_amount) => {
+                    (masked_amount.masked_value, MaskedTokenId::V2(
+                        masked_amount.masked_token_id.clone(),
+                    ), masked_amount.commitment_crc32())
+                },
+        };
```

Comment:
this is what i ended up with after drying it up, but i think maybe it can still be better, like if we add a function it could be:

```
let masked_amount = src.get_masked_amount().map_err(...)?;
let amount_masked_value = masked_amount.get_masked_value();
let commitment = masked_amount.get_commitment();
let amount_masked_token_id = match ... {

};
```